lib/promscrape/discovery/kubernetes/api_watcher.go

```diff
@@ -511,6 +529,7 @@ func newURLWatcher(role, apiURL string, gw *groupWatcher) *urlWatcher {
 }
 
 func (uw *urlWatcher) subscribeAPIWatcherLocked(aw *apiWatcher) {
+	uw.refCount++
```

Comment:
I'm a bit confused with unconditional reference increase/decrease... Is there a guarantee that there are no `subscribeAPIWatcherLocked` duplicate calls for the same `apiWatcher`? Could it be that `refCount ` will be bigger than actual number of watchers?

Reply:
AFAIK the idea is that `apiWatcher` can be used by multiple `groupWatcher`s if they are discovering the same type of k8s objects. But `groupWatcher` will not know if this is a unique `apiWatcher` or is it shared with other `groupWatcher`s. So in case refs count will only be incremented for unique calls to `subscribeAPIWatcherLocked` it might be lower than an actual amount of references taken by `groupWatcher`s and lead to premature stop of `urlWatcher`.